### ISSUE TITLE
V2 Treasury: `enable` method never allows for adding `LIQUIDITY_TOKEN` and `RESERVE_TOKEN` to the registry

### ISSUE THREAD

**Issue by BarnabyBob** [OlympusDAO/olympus-contracts#139]
@olyzeus The following lines effectively prohibit the permissioned Governor from adding any liquidity or reserve assets to the treasury whitelist:
https://github.com/OlympusDAO/olympus-contracts/blob/3bb3605195579a76ee6c060927566388470097d5/contracts/Treasury.sol#L312-L322

https://github.com/OlympusDAO/olympus-contracts/blob/3bb3605195579a76ee6c060927566388470097d5/contracts/Treasury.sol#L402-L416

**Comment by JExtor** [comment]
This is correct in terms of `enable`, it pushes the liquidity or reserve token to their registry on L314, but then on L316 it checks if it is a liquidity token OR a reserve token, which it will be. So on L319 it will then delete that from the registry, the address will still have had the bool been set to true for either the liquidity token or reserve token in the mapping `permissions` L66.

For `execute` it is as expected, it will only remove the address from the liquidity token registry IF the address has been set as a reserve token and vice versa. So in this case it will have been added to its registry.

**Comment by BarnabyBob** [comment]
Ah, now I see what you intended for them. However, bear with me because this raises more questions. 

1. So, first of all, if `enable` and `execute` are mutually exclusive, that is...  when OCG is enabled, the former must be used and viceversa, why would they exhibit a different behavior?

2. And if the above is correct:
> This is correct in terms of enable, it pushes the liquidity or reserve token to their registry on L314, but then on L316 it checks if it is a liquidity token OR a reserve token, which it will be. So on L319 it will then delete that from the registry, the address will still have had the bool been set to true for either the liquidity token or reserve token in the mapping permissions L66.

Then, assuming that OCG is enabled, any Reserve or Liquidity tokens added with the current behavior of `enable` would never show up in the `auditReserves` calculations, since the method relies on `RESERVE/LIQUIDITY_TOKEN`s having been added to the `registry`. Is this also correct?
https://github.com/OlympusDAO/olympus-contracts/blob/5a0712b04df861350eafaed00d34375de0354871/contracts/Treasury.sol#L265-L279

Finally, why does the treasury need to have such twisted logic? I understand the need and desire for it to be compact and efficient in terms of APIs, but at this level it makes it hard to remember how to use policy actions properly!

**Comment by BarnabyBob** [comment]
@lolchocotaco maybe this needs re-opening?

**Comment by BarnabyBob** [comment]
@ind-igo I wouldn't want to open another issue for this, but let's also discuss the scenario where:
- OCG is disabled
- A liquidity token (LT) has been added with  `queueTimelock/execute`
- The newly added LT turns out to be problematic (e.g. a security incident) and needs disabling asap

in the case, where the LT is already in the registry, we would only have `disable` at our disposal to revoke its LT status in `permissions`, however it is currently impossible to erase it from the registry, once it was added.

This has repercussions on `auditReserves`, the method responsible for consolidating treasury reverses and keeping them updated (and in turn on the treasury reserves themselves obviously). So, now we have a problematic or dangerous LT upon which we can act, at best, in two ways to patch:
- act on its bonding calculator, if possible, and adjust it to either return 0 or whatever is safer or more appropriate
- update the `bondCalculator` address associated with the LT to point to a new bonding calculator

In any case and with either fix, the problematic token will still stay on in the registry forever and it will continue to require being accounted for when auditing reserves, even if its `permissions` have long been disabled.

The same problem is present, in the same scenario, with a reserve token (RT) except it's actually worse, since there isn't even that manuverability space made available by the LT calculator configuration (i.e. `bondCalculator`). In this case, it seems impossible to prevent the troubled RT from ever being excluded in `auditReserves`.

Aside from this and going back to the original issue, this wouldn't even with `enable` since neither RTs nor LTs never make it into the registry in the first place, so `auditingReserves` has never anything to cycle through.
